network/p2p/streams.go

```diff
@@ -73,7 +73,17 @@ func (n *streamManager) streamHandler(stream network.Stream) {
 				n.log.Infof("Failed to check old stream with %s: %v", remotePeer, err)
 			}
 			n.streams[stream.Conn().RemotePeer()] = stream
-			n.handler(n.ctx, remotePeer, stream, true)
+
+			// streamHandler is supposed to be called for accepted streams, so we expect incoming here
+			incoming := stream.Conn().Stat().Direction == network.DirInbound
+			if !incoming {
+				if stream.Stat().Direction == network.DirUnknown {
+					n.log.Warnf("Unknown direction for a steam %s to/from %s", stream.ID(), remotePeer)
+				} else {
+					n.log.Warnf("Unexpected outgoing stream in streamHandler for connection %s (%s): %s vs %s stream", stream.Conn().ID(), remotePeer, stream.Conn().Stat().Direction, stream.Stat().Direction.String())
```

Comment:
I was debugging connectivity issues and added these. Never seen DirUnknown but rarely seen the "outgoing" log there but did not debug since the original issue got resolved by that time